File: eCommerce/eCommerce.Services/Database/Reward.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCommerce.Services.Database
{
    public class Reward
    {
        [Key]
        public int Id { get; set; }
        
        public int? RestaurantId { get; set; }
        
        [ForeignKey("RestaurantId")]
        public Restaurant? Restaurant { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Title { get; set; } = string.Empty;
        
        [MaxLength(500)]
        public string? Description { get; set; }
        
        public int PointsRequired { get; set; }
        
        [MaxLength(50)]
        public string? RewardType { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation properties
        public ICollection<UserReward> UserRewards { get; set; } = new List<UserReward>();
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/reward.dart
class Reward {
  final int id;
  final String title;
  final String? description;
  final int pointsRequired;
  final int? restaurantId;
  final bool isActive;
  final DateTime createdAt;
  final int timesClaimed;
  final bool canRedeem;

  Reward({
    required this.id,
    required this.title,
    this.description,
    required this.pointsRequired,
    this.restaurantId,
    required this.isActive,
    required this.createdAt,
    required this.timesClaimed,
    required this.canRedeem,
  });

  factory Reward.fromJson(Map<String, dynamic> json) {
    return Reward(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'],
      pointsRequired: json['pointsRequired'] ?? 0,
      restaurantId: json['restaurantId'],
      isActive: json['isActive'] ?? true,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
      timesClaimed: json['timesClaimed'] ?? 0,
      canRedeem: json['canRedeem'] ?? false,
    );
  }
}
